pkg/innerring/contracts.go

```diff
@@ -121,11 +145,61 @@ func parseAlphabetContracts(cfg *viper.Viper, morph *client.Client) (alphabetCon
 	return alpha, nil
 }
 
-func parseContract(cfg *viper.Viper, morph *client.Client, cfgName, nnsName string) (res util.Uint160, err error) {
-	contractStr := cfg.GetString(cfgName)
-	if len(contractStr) == 0 {
-		return morph.NNSContractAddress(nnsName)
+type nnsContext struct {
+	context.Context
+
+	nnsDeployed bool
+}
+
+func parseContract(ctx *nnsContext, _logger *logger.Logger, cfg *viper.Viper, morph *client.Client, cfgName, nnsName string) (res util.Uint160, err error) {
+	if cfg.IsSet(cfgName) {
+		_logger.Info("decoding configured contract...", zap.String("config key", cfgName))
+		return util.Uint160DecodeStringLE(cfg.GetString(cfgName))
+	}
+
+	const pollInterval = 5 * time.Second // no too long and not so frequently
+
+	if !ctx.nnsDeployed {
+		for {
+			_logger.Info("waiting for NNS contract to be deployed...")
+
+			select {
+			case <-ctx.Done():
+				return res, fmt.Errorf("waiting for NNS contract: %w", ctx.Err())
+			default:
+			}
+
+			_, err = morph.NNSHash()
+			if err == nil {
+				ctx.nnsDeployed = true
+				break
+			}
+
+			// FIXME: unstable approach, but currently there is no other way
+			//  Track https://github.com/neo-project/proposals/pull/156 and its support in
```

Comment:
I think we need some issue in neofs-node to track these places and fix them when it's fixed in some NeoGo release.

Reply:
yeah I'll create one